doc/acl/Everflow-test-plan.md

```diff
@@ -0,0 +1,1157 @@
+        "EF_EGRESS|RULE_4": {
+            "L4_DST_PORT": "4661",
+            "MIRROR_EGRESS_ACTION": "session_1",
+            "PRIORITY": "9996"
+        },
+        "EF_EGRESS|RULE_5": {
+            "ETHER_TYPE": "4660",
+            "MIRROR_EGRESS_ACTION": "session_1",
+            "PRIORITY": "9995"
+        },
+        "EF_EGRESS|RULE_6": {
+            "IP_PROTOCOL": "126",
+            "MIRROR_EGRESS_ACTION": "session_1",
+            "PRIORITY": "9994"
+        },
+        "EF_EGRESS|RULE_7": {
+            "MIRROR_EGRESS_ACTION": "session_1",
+            "PRIORITY": "9993",
+            "TCP_FLAGS": "0x12/0x12"
+        },
+        "EF_EGRESS|RULE_8": {
+            "L4_SRC_PORT_RANGE": "4672-4681",
+            "MIRROR_EGRESS_ACTION": "session_1",
+            "PRIORITY": "9992"
+        },
+        "EF_EGRESS|RULE_9": {
+            "L4_DST_PORT_RANGE": "4672-4681",
+            "MIRROR_EGRESS_ACTION": "session_1",
+            "PRIORITY": "9991"
+        },
+        "EF_EGRESS|RULE_10": {
+            "DSCP": "51",
+            "MIRROR_EGRESS_ACTION": "session_1",
+            "PRIORITY": "9990"
+        },
+        "EF_EGRESS|RULE_11": {
+            "IN_PORTS": "Ethernet4,Ethernet8",
+            "MIRROR_EGRESS_ACTION": "session_1",
+            "PRIORITY": "9989"
+        }
+    }
+}
+```
+
+### Run test
+
+For each configuration scenario, we need to run all the sub-tests. Everflow sub-tests consists of a number of test cases. Each of the test case is executed with log analyzer enabled, for example:
+
+1. Run loganalyzer 'init' phase
+2. Run a everflow test case
+3. Run loganalyzer 'analyze' phase
+
+Each test case may involve with with one or more classes defined in the PTF script.
+
+#### PTF Test
+
+The everflow test cases eventually call the ptf scripts to do the actual testing. The PTF scripts inject packets into DUT and validate traffic forwarded by DUT.
+
+PTF test will generate traffic between ports and make sure it mirrored according to the configured Everflow session and ACL rules. Depending on the testbed topology and the existing configuration (e.g. ECMP, LAGS, etc) packets may arrive to different ports. Therefore ports connection information will be generated from the minigraph and supplied to the PTF script.
+
+The `EverflowTest` class in everflow_tb_test.py need to be extended to cover IPv6 testing. Need some new methods for sending and validating IPv6 packets.
+
+## Test cases
+
+Each test case will be additionally validated by the loganalyzer utility.
+
+Each test case will add dynamic Everflow ACL rules at the beginning and remove them at the end.
+
+Each test case will run traffic for persistent and dynamic Everflow ACL rules.
+
+Each test case will analyze Everflow packet header and payload (if mirrored packet is equal to original). In case of egress mirroring, verify that TTL of the mirrored packet in GRE tunnel is decremented comparing with the injected packet.
+
+### Test case \#1 - Resolved route
```

Comment:
I notice that some of the tests are pretty similar to each other; is it possible to merge some of the tests into single ones so that we could have a simplified list of tests.

Reply:
Agree. I have merged some of the test cases. Now we totally have 4 test cases.